### ISSUE TITLE
Look into using Prettier to auto-format code.

### ISSUE THREAD

**Issue by DamianMullins** [justeat/fozzie#57]
[Prettier](https://prettier.io/docs/en/) "_removes all original styling and ensures that all outputted code conforms to a consistent style_". 

There is a [blog post here](https://jlongster.com/A-Prettier-Formatter) which goes into the reasons why the project was created and the problems it solves. 

It runs seamlessly with any eslint rules already in place and should help us to avoid checking in code which fails the lint checks.

## Example

I've [tested adding Prettier to the `f-validate` module](https://github.com/justeat/f-validate/compare/prettier?expand=1), as you can see from the diff it is quite opinionated, however, the benefit is that we would no longer have to worry about checking for code style errors when reviewing PRs, thereby reducing the amount the of time it takes to perform PR reviews.

**Comment by ashleynolan** [comment]
Agree on putting this in – was thinking about doing this at some point so good that you've done it already! 🥇 

It would be good if we could add this into the gulp-build dependencies rather than into each individual project – guessing we could use something like [gulp-prettier](https://www.npmjs.com/package/gulp-prettier) and then it would just get included and run as part of every module auto-magically?

**Comment by ashleynolan** [comment]
Would also be good to make the rules less harsh as well – unless we want it to do stuff like deleting lines at the start/end of functions and that kind of thing.

Might just mean we need to tighten up the linting rules.

**Comment by DamianMullins** [comment]
Talk from James Long (creator of Prettier) — https://www.youtube.com/watch?v=hkfBvpEfWdA

**Comment by ashleynolan** [comment]
I took a look at Prettier a little last week and tried applying it to one of our modules.

I like what it sets out to achieve, but unless you agree with all of the rules, some of them can be a little grating (as it's intentionally) non-configurable.  ESLInt provides a `fix` flag which would allow us to achieve a pretty similar result, but be more configurable to how we want to style our code as a team.

I've switched this flag on now in `gulp-build-fozzie` and will look to add some more opinionated rules into our ESLint config around code styling (things like removing empty spacing from the start and end of functions).  I'll aim to get this done by the next UI Guild so we can discuss these changes and turn them on for all projects then shortly after, if that sounds like a good plan?
